### PR TITLE
ROC-4262 Use docker in the pipeline

### PR DESCRIPTION
**Pull request by timja**
https://tools.hmcts.net/jira/browse/ROC-4262

Enable docker in the pipeline and AKS

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ ] No
```

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -1,11 +1,11 @@
 FROM openjdk:8-jre-alpine
```
Comment (timwebster9) [inline Dockerfile]: please use the cnp-java-base image: https://github.com/hmcts/cnp-java-base

File: src/kubernetes/.lint.yml [hunk 2]
```diff
@@ -0,0 +1,2 @@
+rules:
+  line-length: disable
```
Comment (timwebster9) [inline src/kubernetes/.lint.yml]: how is this used?  Could we add this to the CI somehow?

File: src/kubernetes/deployment.template.yaml [hunk 3]
```diff
@@ -0,0 +1,161 @@
...
+              value: "${STAFF_EMAIL}"
+            - name: CLAIM_STORE_TEST_SUPPORT_ENABLED
+              value: 'true'
+            - name: FEATURE_TOGGLES_EMAILTOSTAFF
+              value: 'true'
+            - name: ROOT_APPENDER
+              value: 'CMC'
+            - name: APPINSIGHTS_INSTRUMENTATIONKEY
+              value: 'TODO'
+
```
Comment (timwebster9) [inline src/kubernetes/deployment.template.yaml]: You could add a configmap to this file and reference all variables from that, in preparation for https://tools.hmcts.net/jira/browse/CNP-758 if we go down that route

File: src/kubernetes/.lint.yml (hunk 2 above)
Comment (timja) [reply src/kubernetes/.lint.yml]: currently locally with:
`yamllint -c src/kubernetes/.lint.yml src/kubernetes/deployment.template.yaml`

File: src/kubernetes/deployment.template.yaml (hunk 3 above)
Comment (timja) [reply src/kubernetes/deployment.template.yaml]: happy to do it when the library supports it

File: src/kubernetes/.lint.yml (hunk 2 above)
Comment (timwebster9) [reply src/kubernetes/.lint.yml]: ok so it's an external tool? Maybe we could add it to the docker image that kubectl/az cli run in, and run it before the `kubectl apply`.  Only if it adds value though - it's a bit late to run it at this point.

Don't really want to get into having to add stuff to the agent VM and all that :-)

Comment (timja) [reply src/kubernetes/.lint.yml]: should just be able to run it in an image, it should run in the test section i think well before this bit if the flag is enabled. but we can probably leave that for this PR

File: src/kubernetes/deployment.template.yaml (hunk 3 above)
Comment (timwebster9) [reply src/kubernetes/deployment.template.yaml]: if it's part of this same file it'll just work.  See here for example: https://github.com/hmcts/jenkins-container-config/blob/65aac84384bd3e4b45f1027349b682b3f3def387/aks/jenkins.yaml#L31

Comment (timwebster9) [reply src/kubernetes/deployment.template.yaml]: can move it to another file once the library is updated to use different files per env

**Pull request merged**